### PR TITLE
JENKINS-63071 adding repo browser to multibranch pipeline. Resaving c…

### PR DESCRIPTION
**Pull request by dkjellin**
Fix for JENKINS-63071
Fix missing repository browser to multi branch projects.
A re-save of the config is needed to enable the repository browser for existing projects

### CODE REVIEW

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java [hunk 1]
```diff
@@ -116,13 +118,18 @@ public BitbucketSCMSource(
...
             } catch (MirrorFetchException ex) {
                 setEmptyRepository(credentialsId, sshCredentialsId, projectName, repositoryName, serverId, mirrorName);
             }
         } else {
             BitbucketRepository localRepo = scmHelper.getRepository(projectName, repositoryName);
             setRepositoryDetails(credentialsId, sshCredentialsId, serverId, "", localRepo);
+            selfLink = localRepo.getSelfLink();
         }
+        //self link contains `/browse` which we must trim off.
+        String repositoryUrl = selfLink.substring(0, max(selfLink.indexOf("/browse"), 0));
```
Comment (mhenschke-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:131]: /supernit I think we want `selfLink.lastIndexOf("/browse")`, in the case the repo or project name is also browse

**Review by mhenschke-atl — APPROVED** [review]

File: readme.md [hunk 2]
```diff
@@ -212,6 +212,8 @@ Integration tests are run under the `it` profile with the Failsafe plugin using
 ### 2.1.4 (XX XXX 2021)
 - The minimum version of Jenkins changed to be **2.204.4**
 - A number of dependencies upgrades
+- Fix for JENKINS-63071, we are now setting the repository browser also for multibranch projects. Existing projects will
```
[the remark this answers is not in the document]
Comment (dkjellin) [reply readme.md:215]: Yep, I plan to, as soon as this is merged.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java (hunk 1 above)
Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:131]: I copied this from the `BitbucketSCM` happy to change there as well if you think it worth it?

**Review by EliSalter-atl — APPROVED** [review]

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java (hunk 1 above)
Comment (mhenschke-atl) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:131]: /shrug yeah given it's such a simple fix

Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java]: Yep, I've pushed the change, once builds are green I'll merge

**Pull request merged**